lib/enum/arrays.rb

```diff
@@ -1,24 +1,32 @@
 class Array
 
   def sum_of_numeric_elements
+    # inject(0){ |memo, element| element.is_a? Numeric ? (memo + element) : element }
+    select { |element| element.is_a? Numeric }.inject(:+)
   end
 
   def product_of_numeric_elements
+    select{ |element| element.is_a? Numeric }.inject(:*)
   end
 
   def even_numeric_elements
+    select{ |element| element.is_a?(Fixnum) && (element % 2 == 0)}
   end
 
   def odd_numeric_elements
+    select{ |element| element.is_a?(Fixnum) && (element % 2 != 0)}
   end
 
   def string_elements
+    select{ |element| element.is_a?(String) }
   end
 
   def lowercase_string_elements_as_uppercase
+    string_elements.map{ |element| element == element.downcase ? element.capitalize : next }.compact
```

Comment:
Interesting use of `next` here. I think best practice is to use `nil` though, which gives the same result.